File: ChatGPT(예진)/Python (수정 완료)/calendar_program.py
import calendar

def main():
    while True:
        print("1. Select year")
        print("2. Select month")
        print("3. Exit")
        choice = input("Please select: ")

        if choice == "1":
            year = int(input("Enter the year: "))
            month = None
            day = None
            show_month_calendar(year, month, day)
        elif choice == "2":
            year = int(input("Enter the year: "))
            month = int(input("Enter the month: "))
            day = None
            show_day_calendar(year, month, day)
        elif choice == "3":
            print("Exiting the program.")
            break
        else:
            print("Invalid choice. Please choose again.")

def show_month_calendar(year, month, day):
    print("Select a month:")
    for i, month_name in enumerate(calendar.month_name[1:], start=1):
        print(f"{i}. {month_name}")

    month = int(input("Please select: "))
    if month < 1 or month > 12:
        print("Invalid input.")
        return

    _, days_in_month = calendar.monthrange(year, month)
    print(calendar.month(year, month))
    choose_day(year, month, days_in_month)

def show_day_calendar(year, month, day):
    print("Select a date:")
    _, days_in_month = calendar.monthrange(year, month)
    for day in range(1, days_in_month + 1):
        print(day)

    day = int(input("Please select: "))
    if day < 1 or day > days_in_month:
        print("Invalid input.")
        return

    print(f"{year} {month} {day}")
    print(f"Birth flower: {get_birth_flower(month)}")

def choose_day(year, month, days_in_month):
    day = int(input("Select a date: "))
    if day < 1 or day > days_in_month:
        print("Invalid input.")
        return

    print(f"{year} {month} {day}")
    print(f"Birth flower: {get_birth_flower(month)}")

def get_birth_flower(month):
    birth_flowers = {
        1: "Carnation",
        2: "Snowdrop",
        3: "Daffodil",
        4: "Daisy",
        5: "Lily",
        6: "Rose",
        7: "Water Lily",
        8: "Sunflower",
        9: "Poppy",
        10: "Marigold",
        11: "Chrysanthemum",
        12: "Poinsettia"
    }
    return birth_flowers.get(month, "Unknown")

if __name__ == "__main__":
    main()

File: ChatGPT(예진)/Cpp (수정 완료)/calendar_program.cpp
#include <iostream>
#include <string>

#define MAX_MONTH_NAME_LENGTH 20

void show_month_calendar(int year, int month, int day);
void show_day_calendar(int year, int month, int day);
void choose_day(int year, int month, int days_in_month);
const std::string get_birth_flower(int month);

int main() {
    std::string choice;
    int year, month, day;

    while (true) {
        std::cout << "1. Select year\n";
        std::cout << "2. Select month\n";
        std::cout << "3. Exit\n";
        std::cout << "Choose: ";
        std::cin >> choice;

        if (choice == "1") {
            std::cout << "Enter the year: ";
            std::cin >> year;
            month = 0;
            day = 0;
            show_month_calendar(year, month, day);
        }
        else if (choice == "2") {
            std::cout << "Enter the year: ";
            std::cin >> year;
            std::cout << "Enter the month: ";
            std::cin >> month;
            day = 0;
            show_day_calendar(year, month, day);
        }
        else if (choice == "3") {
            std::cout << "Exiting the program.\n";
            break;
        }
        else {
            std::cout << "Invalid choice. Please choose again.\n";
        }
        std::cin.ignore(std::numeric_limits<std::streamsize>::max(), '\n'); // clear input buffer
    }

    return 0;
}

void show_month_calendar(int year, int month, int day) {
    std::cout << "Choose a month:\n";
    std::string month_names[12] = {
        "January", "February", "March", "April", "May", "June", "July",
        "August", "September", "October", "November", "December"
    };

    for (int i = 0; i < 12; i++) {
        std::cout << i + 1 << ". " << month_names[i] << '\n';
    }

    std::cout << "Choose: ";
    std::cin >> month;
    if (month < 1 || month > 12) {
        std::cout << "Invalid input.\n";
        return;
    }

    int days_in_month;
    if (month == 2) {
        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            days_in_month = 29;
        }
        else {
            days_in_month = 28;
        }
    }
    else if (month == 4 || month == 6 || month == 9 || month == 11) {
        days_in_month = 30;
    }
    else {
        days_in_month = 31;
    }

    std::cout << "Calendar for " << year << ' ' << month_names[month - 1] << '\n';
    choose_day(year, month, days_in_month);
}

void show_day_calendar(int year, int month, int day) {
    std::cout << "Choose a date:\n";
    int days_in_month;
    if (month == 2) {
        if ((year % 4 == 0 && year % 100 != 0) || (year % 400 == 0)) {
            days_in_month = 29;
        }
        else {
            days_in_month = 28;
        }
    }
    else if (month == 4 || month == 6 || month == 9 || month == 11) {
        days_in_month = 30;
    }
    else {
        days_in_month = 31;
    }

    for (int i = 1; i <= days_in_month; i++) {
        std::cout << i << '\n';
    }

    std::cout << "Choose: ";
    std::cin >> day;
    if (day < 1 || day > days_in_month) {
        std::cout << "Invalid input.\n";
        return;
    }

    std::cout << year << " year " << month << " month " << day << " day\n";
    std::cout << "Birth flower: " << get_birth_flower(month) << '\n';
}

void choose_day(int year, int month, int days_in_month) {
    int day;
    std::cout << "Choose a date: ";
    std::cin >> day;
    if (day < 1 || day > days_in_month) {
        std::cout << "Invalid input.\n";
        return;
    }

    std::cout << year << " year " << month << " month " << day << " day\n";
    std::cout << "Birth flower: " << get_birth_flower(month) << '\n';
}

const std::string get_birth_flower(int month) {
    const std::string birth_flowers[] = { "", "Poppy", "Primrose", "Dandelion", "Cherry Blossom", "Rose", "Lavender", "Lotus", "Sunflower", "Lily", "Marigold", "Carnation", "Poinsettia" };
    return birth_flowers[month];
}
